packages/cli/src/commands/remove-plugin/removePlugin.test.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fse from 'fs-extra';
+import path from 'path';
+import { paths } from '../../helpers/paths';
+import { addExportStatement, capitalize } from '../create-plugin/createPlugin';
+import { addCodeownersEntry } from '../create-plugin/lib/codeowners';
+import {
+  removeReferencesFromAppPackage,
+  removeReferencesFromPluginsFile,
+  removePluginDirectory,
+  removeSymLink,
+  removePluginFromCodeOwners,
+} from './removePlugin';
+
+const BACKSTAGE = `@backstage`;
+const testPluginName = 'yarn-test-package';
+const testPluginPackage = `${BACKSTAGE}/plugin-${testPluginName}`;
+
+const removeEmptyLines = (file: string): string =>
+  file
+    .split('\n')
+    .filter(Boolean)
+    .join('\n');
+
+const createTestPackageFile = async (
+  testFilePath: string,
+  packageFile: string,
+) => {
+  // Copy contents of package file for test
+  const packageFileContent = JSON.parse(fse.readFileSync(packageFile, 'utf8'));
+
+  packageFileContent.dependencies[testPluginPackage] = '0.1.0';
+  fse.createFileSync(testFilePath);
+  fse.writeFileSync(
+    testFilePath,
+    `${JSON.stringify(packageFileContent, null, 2)}\n`,
+    'utf8',
+  );
+  return;
+};
+const createTestPluginFile = async (
+  testFilePath: string,
+  pluginsFilePath: string,
+) => {
+  // Copy contents of package file for test
+  fse.copyFileSync(pluginsFilePath, testFilePath);
+  const pluginNameCapitalized = testPluginName
+    .split('-')
+    .map(name => capitalize(name))
+    .join('');
+  const importStatement = `import { default as ${pluginNameCapitalized}} from @backstage/plugin-${testPluginName}`;
+  const exportStatement = `export {${pluginNameCapitalized}}`;
+  addExportStatement(testFilePath, importStatement, exportStatement);
+};
+
+function mkTestDir(testDirPath: string) {
+  fse.mkdirSync(testDirPath);
+  for (let i = 0; i < 50; i++)
+    fse.createFileSync(path.join(testDirPath, `testFile${i}.ts`));
+}
+
+describe('removePlugin', () => {
+  describe('Remove Plugin Dependencies', () => {
+    const appPath = paths.resolveTargetRoot('packages', 'app');
+    const githubDir = paths.resolveTargetRoot('.github');
+    it('removes plugin references from /packages/app/package.json', async () => {
+      // Set up test
+      const packageFilePath = path.join(appPath, 'package.json');
+      const testFilePath = path.join(appPath, 'test.json');
```

Comment:
Not crazy about the tests reading/writing to the repo itself. Would prefer things being tested in a temporary directory. And also move to fs mock eventually :p